src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -352,39 +358,70 @@ public ResponseEntity<ProgrammingExerciseStudentParticipation> resumeParticipati
     @PutMapping("exercises/{exerciseId}/request-feedback")
     @EnforceAtLeastStudent
     @FeatureToggle(Feature.ProgrammingExercises)
-    public ResponseEntity<ProgrammingExerciseStudentParticipation> requestFeedback(@PathVariable Long exerciseId, Principal principal) {
+    public ResponseEntity<StudentParticipation> requestFeedback(@PathVariable Long exerciseId, Principal principal) {
         log.debug("REST request for feedback request: {}", exerciseId);
-        var programmingExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(exerciseId);
 
-        if (programmingExercise.isExamExercise()) {
-            throw new BadRequestAlertException("Not intended for the use in exams", "participation", "preconditions not met");
+        Exercise exercise = exerciseRepository.findById(exerciseId).orElseThrow(() -> new ResourceNotFoundException("Exercise not found"));
+
+        if (!(exercise instanceof TextExercise) && !(exercise instanceof ProgrammingExercise)) {
+            throw new BadRequestAlertException("Unsupported exercise type", "participation", "unsupported type");
         }
 
-        if (programmingExercise.getDueDate() != null && now().isAfter(programmingExercise.getDueDate())) {
+        return handleExerciseFeedbackRequest(exercise, principal);
+    }
+
+    private ResponseEntity<StudentParticipation> handleExerciseFeedbackRequest(Exercise exercise, Principal principal) {
+        // Validate exercise and timing
+        if (exercise.isExamExercise()) {
+            throw new BadRequestAlertException("Not intended for the use in exams", "participation", "preconditions not met");
+        }
+        if (exercise.getDueDate() != null && now().isAfter(exercise.getDueDate())) {
             throw new BadRequestAlertException("The due date is over", "participation", "preconditions not met");
         }
+        if (exercise instanceof ProgrammingExercise) {
+            ((ProgrammingExercise) exercise).validateSettingsForFeedbackRequest();
+        }
 
-        var participation = programmingExerciseParticipationService.findStudentParticipationByExerciseAndStudentId(programmingExercise, principal.getName());
+        // Get and validate participation
         User user = userRepository.getUserWithGroupsAndAuthorities();
+        StudentParticipation participation = (exercise instanceof ProgrammingExercise)
+                ? programmingExerciseParticipationService.findStudentParticipationByExerciseAndStudentId(exercise, principal.getName())
+                : studentParticipationRepository.findByExerciseIdAndStudentLogin(exercise.getId(), principal.getName())
```

Comment:
Why can't you always use the studentParticipationRepository? The programmingExerciseParticipationService seams to also handle team exercises, why do you not handle them for text exercises as well?